test/addons-napi/test_conversions/test_conversions.c

```diff
@@ -0,0 +1,236 @@
+#include <node_api.h>
+
+void ThrowLastError(napi_env env) {
+  const napi_extended_error_info* error_info = napi_get_last_error_info();
+  if (error_info->error_code != napi_ok) {
+    napi_throw_error(env, error_info->error_message);
+  }
+}
+
+void AsBool(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  napi_value input;
+  status = napi_get_cb_args(env, info, &input, 1);
+  if (status != napi_ok) goto Exit;
+
+  bool value;
+  status = napi_get_value_bool(env, input, &value);
+  if (status != napi_ok) goto Exit;
+
+  napi_value output;
+  status = napi_get_boolean(env, value, &output);
+  if (status != napi_ok) goto Exit;
+
+  status = napi_set_return_value(env, info, output);
+  if (status != napi_ok) goto Exit;
+
+Exit:
+  if (status != napi_ok) ThrowLastError(env);
+}
+
+void AsInt32(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  napi_value input;
+  status = napi_get_cb_args(env, info, &input, 1);
+  if (status != napi_ok) goto Exit;
+
+  int32_t value;
+  status = napi_get_value_int32(env, input, &value);
+  if (status != napi_ok) goto Exit;
+
+  napi_value output;
+  status = napi_create_number(env, value, &output);
+  if (status != napi_ok) goto Exit;
+
+  status = napi_set_return_value(env, info, output);
+  if (status != napi_ok) goto Exit;
+
+Exit:
```

Comment:
Fixed in https://github.com/nodejs/node/pull/11975/commits/9c0b1517b894d0a655377d23f9e4b9943d93aa38